interactions/api/models/misc.py

```diff
@@ -7,58 +7,116 @@
 
 import datetime
 from base64 import b64encode
+from functools import partial, wraps
 from io import FileIO, IOBase
 from logging import Logger
 from math import floor
 from os.path import basename
-from typing import Optional, Union
+from typing import Optional, Tuple, Union
+
+import attrs
 
 from interactions.base import get_logger
 
 log: Logger = get_logger("mixin")
 
 
-class DictSerializerMixin(object):
-    """
-    The purpose of this mixin is to be subclassed.
+class MISSING:
+    """A pseudosentinel based from an empty object. This does violate PEP, but, I don't care."""
 
-    .. note::
-        On subclass, it:
-            -- From kwargs (received from the Discord API response), add it to the `_json` attribute
-            such that it can be reused by other libraries/extensions
-            -- Aids in attributing the kwargs to actual model attributes, i.e. `User.id`
-            -- Dynamically sets attributes not given to kwargs but slotted to None, signifying that it doesn't exist.
-
-    .. warning::
-        This does NOT convert them to its own data types, i.e. timestamps, or User within Member. This is left by
-        the object that's using the mixin.
-    """
+    ...
+
+
+@attrs.define(eq=False, init=False, on_setattr=attrs.setters.NO_OP)
+class DictSerializerMixin:
+    _json: dict = attrs.field(init=False, repr=False)
+    extras: dict = attrs.field(init=False, repr=False)
```

Comment:
`extras` should be underscored *just* in case Discord adds an attribute called just that.

Reply:
Changed in [7e1a139](https://github.com/interactions-py/library/pull/795/commits/7e1a13994ce64c992cc6ea550281bca170f3e6f0)